presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PayloadJoinOptimizer.java

```diff
@@ -238,6 +247,8 @@ public PlanNode visitJoin(JoinNode joinNode, RewriteContext<JoinContext> context
 
             if (isTopJoin && context.get().needsPayloadRejoin()) {
                 PlanNode payloadJoin = transformJoin(newJoinNode, joinContext);
+                context.get().setPayloadNode(null);
```

Comment:
can you explain why the payloadNode gets set to null here?

Reply:
added a comment